addons/mail/static/src/messaging/component/message_list/message_list.js

```diff
@@ -166,23 +166,12 @@ class MessageList extends Component {
     /**
      * @returns {mail.messaging.component.Message|undefined}
      */
-    get lastCurrentPartnerMessageRef() {
-        const currentPartnerMessageRefs = this.messageRefs.filter(messageRef =>
-            (
-                messageRef.message.author &&
-                messageRef.message.author === this.env.messaging.currentPartner
-            )
-        );
-        const { length: l, [l - 1]: lastCurrentPartnerMessageRefs } = currentPartnerMessageRefs;
-        return lastCurrentPartnerMessageRefs;
-    }
-
-    /**
-     * @returns {mail.messaging.component.Message|undefined}
-     */
-    get lastMessageRef() {
-        const { length: l, [l - 1]: lastMessageRef } = this.messageRefs;
-        return lastMessageRef;
+    get mostRecentMessageRef() {
+        if (this.props.order === 'desc') {
+            return this.messageRefs[0];
+        }
+        const { length: l, [l - 1]: mostRecentMessageRef } = this.messageRefs;
+        return mostRecentMessageRef;
```

Comment:
Is that really a recommended syntax in the JS world? cc @alexkuhn 

Feels like the normal syntax is shorter and more readable for everyone?
```suggestion
        return this.messageRefs[this.messageRefs.length - 1];
```

Reply:
Both syntax are valid, each has pros/cons.

Your suggestion has less chars, but its quite a ragbag of an expression (`this.messageRefs` is repeated twice, expression inside array index access is not great from a readability PoV, etc...), and doesn't hint at what we want to get (here the most recent message ref).

The other one is longer, but more expressive in my opinion, but that's more a preference I guess.

Reply:
We would have a consensus if arrays had a method `.last()`

Reply:
Shortest and no repeat: `return this.messageRefs.slice(-1)[0]` ?

Reply:
This is just for fun btw, I don't really care either way. If all else is good we can merge it.

Reply:
good suggestion :)

Reply:
But I have read it is much slower than accessing by index, so probably not so good. :(

Reply:
1 less!
`return [...this.messageRefs].pop()]`
But let's merge before this goes too far?

Reply:
I'll leave it like that hein, ok :D ?

Reply:
It is of the utmost importance to apply this change in the whole Odoo codebase!
...just kidding, you can make the last suggested change by seb just for this one.

Reply:
wow, just noticed the history of messages in this thread is messed up, it shows my message "good suggestion :)" twice as 2 different and non-successive occurrences, but when you look at the date it's exactly the same message...

Reply:
nevermind, page reload somehow duplicated messages at random, that's a github bug apparently :p